src/services/activation/services/eMailAddress/index.js

```diff
@@ -0,0 +1,165 @@
+		text: `Bestätige deine E-Mail-Adresse
+\nHallo ${user.firstName},
+\nbitte bestätige deine neue E-Mail-Adresse (${email}) über folgenden Link: ${activationLink}
+\nBitte beachte, dass der Aktivierungslink nur 2 Stunden gültig ist.
+\nDein ${SC_SHORT_TITLE} Team`,
+		html: '',
+	};
+
+	return new Mail(subject, content, email);
+};
+
+const buildFYIMail = (user) => {
+	/* eslint-disable max-len */
+	const subject = 'E-Mail-Adresse geändert';
+	const content = {
+		text: `E-Mail-Adresse geändert
+\nHallo ${user.firstName},
+\nwir wollten dich nur informieren, dass sich die E-Mail-Adresse für dein ${SC_SHORT_TITLE} Konto geändert hat.
+\nWenn du die Änderung veranlasst hast, ist alles in Ordnung.
+\nFalls du nicht darum gebeten hast, deine E-Mail-Adresse zu änderen, kontaktiere deinen Schuladministrator oder unseren User-Support.
+\nDein ${SC_SHORT_TITLE} Team`,
+		html: '',
+	};
+
+	return new Mail(subject, content, user.email);
+};
+
+const mail = async (ref, type, user, entry) => {
+	let content;
+	switch (type) {
+		case 'activationLinkMail':
+			content = buildActivationLinkMail(user, entry);
+			break;
+
+		case 'fyiMail':
+			content = buildFYIMail(user);
+			break;
+
+		default:
+			throw new Error('Mail type not defined');
+	}
+
+	await sendMail(ref, content, entry);
+};
+
+/** This service takes care of what should happen when an activation
+ * code is redeemed, with the keyword eMailAdress. In addition,
+ * this service can be used to create an job to change the email/username.
+ */
+class EMailAddressActivationService {
+	/**
+	 * create job
+	 */
+	async create(data, params) {
+		if (!data || !data.email || !data.password) throw new BadRequest('Missing information');
+		const user = await getUser(this, params.account.userId);
+
+		// create new entry
+		const entry = await createEntry(this, params.account.userId, E_MAIL_ADDRESS, data.email);
+
+		// send email
+		await mail(this, 'activationLinkMail', user, entry);
+		return { success: true };
+	}
+
+	async update(id, data, params) {
+		const { entry, user } = data;
+		const account = await this.app.service('/accounts').find({
+			query: {
+				userId: user._id,
+			},
+		});
+		if ((account || []).length !== 1) throw new Forbidden(customErrorMessages.NOT_AUTHORIZED);
+
+		const email = entry.quarantinedObject;
+		if (!email) {
+			await deleteEntry(this, entry._id);
+			throw new GeneralError('Link incorrectly constructed and will be removed');
+		}
+
+		try {
+			await setEntryState(this, entry._id, STATE.PENDING);
+
+			// update user and account
+			await this.app.service('users').patch(account[0].userId, { email });
+			await this.app.service('/accounts').patch(account[0]._id, { username: email });
+
+			// set activation link as consumed
+			await setEntryState(this, entry._id, STATE.SUCCESS);
+		} catch (error) {
+			logger.error(error);
+			await setEntryState(this, entry._id, STATE.ERROR);
+		}
+
+		// send fyi mail to old email
+		await mail(this, 'fyiMail', user, entry);
```

Comment:
is this the right moment to send this mail? or should it be done on the create already?

Reply:
Good question. I think it only makes sense here, if something in the account was really changed. If necessary we could send an additional email with different content, on create.
There is already an email sent to the new email address with the activation code

Reply:
We will not chnage this now, perhaps in connection with another ticket